QEfficient/cloud/infer.py

```diff
@@ -6,20 +6,18 @@
 # -----------------------------------------------------------------------------
 
 import argparse
+import logging
 import os
```

Comment:
Some general comments on this file and other HL Apis():

- **Logging**: Mixing logging `(logger.info()) `with direct printing` (print()) `can lead to inconsistent output handling
- **Code Organization**: The `main `function is quite long and handles too many responsibilities. It would benefit from being broken down into smaller, more focused functions. This makes the code easier to read, test, and maintain.
- **Type Annotations**: The use of # type: ignore comments suggests that there might be issues with type checking, can we address the type issues directly, or improve to reflect the code's behavior.

WDYT?

Reply:
Something like below, attaching pseudo code: Whatever you think is best, just to reorganize main to specific focus task and break to make it more readable.
`     

      import QEfficient
      from QEfficient.utils.logging_utils import logger
      
      def compile_qpc(onnx_model_path, qpc_dir_path, **compile_params):
          generated_qpc_path = QEfficient.compile()
            logger.info(f"Compiled qpc files can be found at : {generated_qpc_path}")
            return generated_qpc_path
        
        def handle_existing_files():
            qpc_path_exists, qpc_dir_path = qpc_exists(model_name, qpc_base_dir_name)
            onnx_path_exists, _, _ = onnx_exists(model_name)
  
            if qpc_path_exists:
                logger.info(f"Pre-compiled qpc found at {qpc_dir_path} Executing with given prompt")
            elif onnx_path_exists:
              logger.info(f"Pre-exported ONNX files found at {onnx_dir_path} Jumping to Compilation")
              return compile_qpc(onnx_model_path, qpc_dir_path, **compile_params)
          else:
              (or run normal e2e or raise error)
              raise FileNotFoundError("Neither pre-compiled qpc nor ONNX files were found.")

  
      def main(args):
        # Simplified for brevity; implement actual logic here based on original code structure
        pass
      
      if __name__ == "__main__":
          parser = argparse.ArgumentParser(description="Inference command for Cloud AI 100")
          # Argument definitions remain unchanged
          args = parser.parse_args()
      
          if args.verbose:
              logger.setLevel(logging.INFO)
      
          main(args)
`

Reply:
will create this function, looks good.
In case` onnx/qpc` not found, it will download and do everything till execute.

Reply:
Fixed. please check, I have simplified the flow of infer file, and created two functions
1. for getting qpc dir name moved to utils.py
2. for export -> kept this in export.py as this file also uses it, and did not want to move it to utils cause utils doesn't import anything like transform/export APIs, and wanted to keep it that way.